torch/__init__.py

```diff
@@ -1243,4 +1243,9 @@ def _register_device_module(device_type, module):
 import torch.fx.experimental.symbolic_shapes
 
 from torch import func as func
+
 from torch.func import vmap
+
+# Enable trace collection through dynolog
+from torch.profiler.profiler import profile_using_dynolog
+profile_using_dynolog()
```

Comment:
Why isn't this done in the profiler init itself? That doesn't sound like it should be in the main `__init__` file.

Reply:
The goal of this PR and the optimizer hooks is to profile the code without any code instrumentation about profiling by the user. If we place the profile_using_dynolog call inside the profiler init then it requires the user to import profiler, which beats the intended purpose.